openassessment/xblock/static/js/src/studio/oa_edit.js

```diff
@@ -161,12 +161,16 @@ export class StudioView {
     //
     // The `validate()` method calls `validate()` on any subviews,
     // so that each subview has the opportunity to validate
-    // its fields.
+    // its fields. It returns tabs which fail validation.
     this.clearValidationErrors();
-    if (!this.validate()) {
+    const tabsFailingValidation = this.validate();
+    this.markTabsWithValidationErrors(tabsFailingValidation);
+
+    if (tabsFailingValidation.length > 0) {
+      const tabNames = tabsFailingValidation.map(tab => tab.tabDisplayName);
       this.alert.setMessage(
-        gettext('Couldn\'t Save This Assignment'),
-        gettext('Please correct the outlined fields.'),
```

Comment:
Should we preserve this message?

Reply:
This was an intentional language change from UX so I'm going to say we leave it as coded.